feat(proxy): add max_parallel_requests for model/user/team levels

peterkc opened the pull request:
## Summary

Implements per-model, per-user, and per-team `max_parallel_requests` limits in the proxy rate limiter, addressing three TODOs in `parallel_request_limiter.py`.

- **Model-level**: Configurable via key metadata (`model_max_parallel_requests`) or `model_max_budget`
- **User-level**: New `user_max_parallel_requests` field on `UserAPIKeyAuth`
- **Team-level**: New `team_max_parallel_requests` field on `UserAPIKeyAuth`

## Changes

| File | Change |
|------|--------|
| `auth_utils.py` | Added `get_key_model_max_parallel_requests()` helper |
| `_types.py` | Added `user_max_parallel_requests` and `team_max_parallel_requests` fields |
| `parallel_request_limiter.py` | Wired all 3 TODO locations (v1 limiter) |
| `parallel_request_limiter_v3.py` | Added model-level support to v3 limiter |
| `test_max_parallel_requests.py` | 10 unit tests |

## Documentation

📄 **[Model-Specific Max Parallel Requests](docs/my-website/docs/proxy/model_max_parallel_requests.md)** - New doc with:
- Quick start guide
- Key, team, and priority override examples
- Mermaid flowchart showing priority resolution

## Config Examples

**Model-level** (via API key metadata):
```json
{
  "model_max_parallel_requests": {
    "gpt-4": 5,
    "gpt-3.5-turbo": 20
  }
}
```

**User-level**: Set `user_max_parallel_requests` on user record  
**Team-level**: Set `team_max_parallel_requests` on team record

## Test plan

- [x] Unit tests pass (10/10)
- [x] v3 limiter support added
- [x] Documentation added
- [ ] CI passes

Closes #17824

peterkc committed:
feat(proxy): add max_parallel_requests for model/user/team levels

- Add `get_key_model_max_parallel_requests()` helper in auth_utils.py
- Add `user_max_parallel_requests` and `team_max_parallel_requests` fields to UserAPIKeyAuth
- Wire model-level limits in parallel_request_limiter.py (line 323 TODO)
- Wire user-level limits in parallel_request_limiter.py (line 366 TODO)
- Wire team-level limits in parallel_request_limiter.py (line 394 TODO)
- Add 10 unit tests for helper function and type fields

Closes #17824

JiangJiaWei1103 reviewed (commented):
Over LGTM. I'll test this integration in our use case. Thanks!

peterkc committed:
feat(proxy): add model max_parallel_requests to v3 limiter + docs

- Port model-level max_parallel_requests support to v3 rate limiter
- Add documentation with mermaid priority flowchart
- Covers key-level, team-level, and priority override patterns

krrishdholakia commented:
do it on v3 not v1 @peterkc 

including team level limiting with priority overrides would be great!
